src/actions/heap/heap.ts

```diff
@@ -99,23 +105,45 @@ export class HeapAction extends Hub.Action {
           if (!heapFieldValue) {
             return
           }
-          const requestBody = this.constructBodyForRequest(
-            request.params.heap_env_id!,
-            heapField,
-            heapFieldValue,
-            properties,
-          )
-          req.post({
-            uri: requestUrl,
-            headers: { "Content-Type": "application/json" },
-            body: requestBody,
-          })
+          requestBatch.push({ heapFieldValue, properties })
+          if (requestBatch.length === HeapAction.ROWS_PER_BATCH) {
+            const requestBody = this.constructBodyForRequest(
```

Comment:
extract this to a method since it looks to be duplicated with the code below?

Reply:
if we were to extract this into a `async` method, then the `onRow` callback would have to be made `async` as well. i don't think that's possible because according to the types in `streamJsonDetail`, `onRow` has to return `void`